File: app/src/main/java/co/tpcreative/supersafe/model/HeaderItem.java
package co.tpcreative.supersafe.model;
import androidx.annotation.NonNull;
import java.util.Date;

public class HeaderItem extends ListItem {

	@NonNull
	private Date date;

	public HeaderItem(@NonNull Date date) {
		this.date = date;
	}

	@NonNull
	public Date getDate() {
		return date;
	}

	// here getters and setters
	// for title and so on, built
	// using date

	@Override
	public int getType() {
		return TYPE_HEADER;
	}

	@Override
	public EnumEvent getTypeEvent() {
		return null;
	}
}

File: app/src/main/java/co/tpcreative/supersafe/model/HeaderItem.kt
package co.tpcreative.supersafe.model
import java.util.*

class HeaderItem(private val date: Date) : ListItem() {
    fun getDate(): Date {
        return date
    }

    // here getters and setters
    // for title and so on, built
    // using date
    override fun getType(): Int {
        return ListItem.Companion.TYPE_HEADER
    }

    override fun getTypeEvent(): EnumEvent? {
        return null
    }

}